### PR TITLE
Simplify docconv

### PR DESCRIPTION
**Pull request by pilz**
Hold, this is only to abuse jenkins for a test run. Do not merge!

### CODE REVIEW

**Comment by pilz** [comment]
@gyst @wengole Here is a version that rips out nearly all of docconv. This now is using as much of collective.documentviewer as possible.

I followed Ben's approach to move necessary access code to pi_api and moved also the tests for it there. Now, docconv.client contains also very minimal code to setup and config c.dv, to register our own handler and to provide the indexer that let's us know from the catalog if there are previews available. I bet these can go into other packages and we can remove docconv.client for good.

I would be very happy about a review so that I can start to make the classic workspace upload async.

**Comment by gyst** [comment]
@pilz can you merge master and resolve the conflicts? Would like to review and merge this.

**Comment by pilz** [comment]
done

File: src/ploneintranet/activitystream/browser/prototype/post.html [hunk 1]
```diff
@@ -59,13 +59,34 @@
...
         <img src="/media/preview_thumb_1.jpg" alt=""
              tal:attributes="src string:${attachment/img_src}">
       </a>
+          <figcaption>
+            <strong class="title">
+              <a tal:attributes="href attachment/link; alt attachment/alt" href="/projection-material" alt="Projection Material">
+                <tal:r replace="attachment/title">Projection Material</tal:r> <!--span class="page-counter icon-docs">3x</span-->
+              </a>
+              <!-- This needs exposure of all generated thumbs. We can do that and have them accessible, it is just not added here yet. -->
+              <!--span class="functions">
```
Comment (gyst) [inline src/ploneintranet/activitystream/browser/prototype/post.html:75]: Why is this not exposed, and what needs to be done for that? Can you maybe add a ticket for that?

Comment (pilz) [inline src/ploneintranet/activitystream/browser/prototype/post.html:75]: Cornelis has done additional design for this and I got that in while updating the prototype. I took over the full markup but to implement it, we need to do some tweaks to get more than one preview in. Not difficult, but another "Nebenkriegsschauplatz". At the moment it behaves as before. I consider that a new feature. 

I don't feel qualified to add that ticket as I have not initiated that design change, I frankly don't know more than what I can see from the markup.

Comment (gyst) [inline src/ploneintranet/activitystream/browser/prototype/post.html:75]: OK he talked me through that the other day so I guess this is on my plate ;-).

**Comment by gyst** [comment]
Apart from the fact that this change is too big for me to comprehend (my problem), a few remarks:
- I really really dislike doing actual implementations in ploneintranet.api. If we follow this route we can put everything in ploneintranet.api and don't need any other namespaces anymore. Doing the actual preview logic (and testing those) belongs in ploneintranet.attachments IMO. Especially a massive private function like `_get_dv_data(obj)` destroys the readability of the API.
- It's better to have theme/generated changes in separate pull requests, but of course not a show stopper.

**Comment by pilz** [comment]
Ok, so we could actually move this back into docconv.client and just reference it from the api. I don't think that would be a big issue.

**Comment by pilz** [comment]
jenkins please test this

**Comment by gyst** [comment]
PR build fails because of #731.

**Comment by pilz** [comment]
test please test

**Comment by pilz** [comment]
test please test

**Comment by pilz** [comment]
some obscure github checkout problems are happening

**Comment by pilz** [comment]
@gyst review now while it is still hot :)

**Comment by gyst** [comment]
@pilz waiting for your promised refactorings...

**Comment by pilz** [comment]
ah, ok

**Comment by pilz** [comment]
@gyst the preview code is back in docconv.client now, the api does only call it.

**Pull request merged**